### PR TITLE
Configuration parameter to hide addresses in stack traces

### PR DESCRIPTION
**Pull request by alexey-milovidov**
### Changelog category (leave one):
- Improvement

### Changelog entry (a user-readable short description of the changes that goes to CHANGELOG.md):
Add a configuration parameter to hide addresses in stack traces. It may improve security a little but generally, it is harmful and should not be used.

### CODE REVIEW

**Comment by alexey-milovidov** [comment]
- [ ] clickhouse-local
- [ ] integration test

**Comment by alexey-milovidov** [comment]
Checked manually that it works...

File: src/Common/StackTrace.cpp [hunk 1]
```diff
@@ -30,7 +44,7 @@ std::string signalToErrorMessage(int sig, const siginfo_t & info, [[maybe_unused
             /// Print info about address and reason.
             if (nullptr == info.si_addr)
                 error << "Address: NULL pointer.";
-            else
+            else if (show_addresses.load(std::memory_order_relaxed))
```
Comment (tavplubix) [inline src/Common/StackTrace.cpp:47]: Maybe we can print address anyway if its within the first 4KB? Sometimes it's useful to know offset when debugging segfaults with null pointer

File: programs/server/config.xml [hunk 2]
```diff
@@ -1386,4 +1386,10 @@
         <lru_cache_size>268435456</lru_cache_size>
         <continue_if_corrupted>true</continue_if_corrupted>
     </merge_tree_metadata_cache-->
+
+    <!-- This allows to disable exposing addresses in stack traces for security reasons.
+         Please be aware that it does not improve security much, but makes debugging much harder.
+         I don't recommend to change this setting.
+    -->
+    <show_addresses_in_stack_traces>true</show_addresses_in_stack_traces>
```
Comment (tavplubix) [inline programs/server/config.xml:1394]: Should we add a note about system tables like `stack_traces` and `trace_log` (that this setting use useless without restricting access to system tables)?

**Review by tavplubix — APPROVED** [review]

**Comment by tavplubix** [comment]
Integration tests (asan) [1/3] - #39775

File: programs/server/config.xml [hunk 3]
```diff
@@ -1386,4 +1386,13 @@
...
     </merge_tree_metadata_cache-->
+
+    <!-- This allows to disable exposing addresses in stack traces for security reasons.
+         Please be aware that it does not improve security much, but makes debugging much harder.
+         The addresses that are small offsets from zero will be displayed nevertheless to show nullptr dereferences.
+         Regardless of this configuration, the addresses are visible in the system.stack_trace and system.trace_log tables
+         if the user has access to these tables.
+         I don't recommend to change this setting.
+    -->
+    <show_addresses_in_stack_traces>false</show_addresses_in_stack_traces>
```
Comment (tavplubix) [inline programs/server/config.xml:1397]: I think it was not supposed to be disabled by default (at least we should not disable it in CI), let's comment out this line: https://github.com/ClickHouse/ClickHouse/pull/40749

**Pull request merged**